algorithms/branching.py

```diff
@@ -31,21 +32,24 @@ def count_isomorphisms(G: 'Graph', D: 'List[Vertex]', I: 'List[Vertex]', count_f
             return True
         return 1
 
+    # Start branching algorithm by making a copy of the graph, each branch should have a new copy
     G_copy = G.copy()
     colors, max_colornum = get_colors(G_copy)
 
-    # Choose the color C with at least 4 vertices of that color in the graph
+    # Choose the color class C with at least 4 vertices of that color in the graph
+    # Choose color class with smallest amount of vertices
+    C_len = inf
     for key in colors:
-        if len(colors[key]) >= 4:
+        if len(colors[key]) >= 4 and len(colors[key]) < C_len:
             C = key
-            continue
+            C_len = len(colors[key])
 
     # Choose the first occurring vertex with color C in the list of vertices of the first graph
-    g0 = []
     for v in colors[C]:
```

Comment:
Kan het voorkomen dat de variabele C niet geinitialiseerd wordt ?

Reply:
Ik had even niet gekeken naar de functies die ervoor worden uitgevoerd.
C wordt altijd geset omdat de graph wel balanced, maar niet bijected is.
- resolved -